File: cpp/Settings/InGameTitanSettings.c
// Type: Settings.InGameTitanSettings
// Ghidra (System V/gcc) decompilation, IL2CPP structs + signatures applied.
// ---- AoTTG2 cross-reference ----
// Update status: unchanged since the previous dump
// C# structure: source/csharp/Scripts/Settings/InGameTitanSettings.cs
// Prior real C# source (older reference): Assets/Scripts/Settings/InGame/InGameTitanSettings.cs
// --------------------------------

// Settings.InGameTitanSettings$$.ctor
// il2cpp: void Settings_InGameTitanSettings___ctor (Settings_InGameTitanSettings_o* __this, const MethodInfo* method);
// 0x408bff0

void Settings_InGameTitanSettings___ctor(Settings_InGameTitanSettings_o *__this,MethodInfo *method)

{
  Settings_BoolSetting_o *pSVar1;
  Settings_FloatSetting_o *pSVar2;
  Settings_IntSetting_o *pSVar3;
  
  if (g_data_057ac3f8 == '\0') {
    il2cpp_runtime_helper_023445d0(&TypeInfo_BoolSetting);
    il2cpp_runtime_helper_023445d0(&TypeInfo_FloatSetting);
    il2cpp_runtime_helper_023445d0(&TypeInfo_IntSetting);
    g_data_057ac3f8 = '\x01';
  }
  pSVar1 = (Settings_BoolSetting_o *)il2cpp_runtime_helper_023052d0(TypeInfo_BoolSetting);
  Settings_BoolSetting___ctor_40f3960(pSVar1,0,(MethodInfo *)0x0);
  (__this->fields).TitanSpawnEnabled = pSVar1;
  il2cpp_runtime_helper_022b4080(&(__this->fields).TitanSpawnEnabled);
  pSVar2 = (Settings_FloatSetting_o *)il2cpp_runtime_helper_023052d0(TypeInfo_FloatSetting);
  Settings_FloatSetting___ctor_40f3a30(pSVar2,20.0,0.0,100.0,(MethodInfo *)0x0);
  (__this->fields).TitanSpawnNormal = pSVar2;
  il2cpp_runtime_helper_022b4080(&(__this->fields).TitanSpawnNormal);
  pSVar2 = (Settings_FloatSetting_o *)il2cpp_runtime_helper_023052d0(TypeInfo_FloatSetting);
  Settings_FloatSetting___ctor_40f3a30(pSVar2,20.0,0.0,100.0,(MethodInfo *)0x0);
  (__this->fields).TitanSpawnAbnormal = pSVar2;
  il2cpp_runtime_helper_022b4080(&(__this->fields).TitanSpawnAbnormal);
  pSVar2 = (Settings_FloatSetting_o *)il2cpp_runtime_helper_023052d0(TypeInfo_FloatSetting);
  Settings_FloatSetting___ctor_40f3a30(pSVar2,15.0,0.0,100.0,(MethodInfo *)0x0);
  (__this->fields).TitanSpawnJumper = pSVar2;
  il2cpp_runtime_helper_022b4080(&(__this->fields).TitanSpawnJumper);
  pSVar2 = (Settings_FloatSetting_o *)il2cpp_runtime_helper_023052d0(TypeInfo_FloatSetting);
  Settings_FloatSetting___ctor_40f3a30(pSVar2,15.0,0.0,100.0,(MethodInfo *)0x0);
  (__this->fields).TitanSpawnCrawler = pSVar2;
  il2cpp_runtime_helper_022b4080(&(__this->fields).TitanSpawnCrawler);
  pSVar2 = (Settings_FloatSetting_o *)il2cpp_runtime_helper_023052d0(TypeInfo_FloatSetting);
  Settings_FloatSetting___ctor_40f3a30(pSVar2,15.0,0.0,100.0,(MethodInfo *)0x0);
  (__this->fields).TitanSpawnThrower = pSVar2;
  il2cpp_runtime_helper_022b4080(&(__this->fields).TitanSpawnThrower);
  pSVar2 = (Settings_FloatSetting_o *)il2cpp_runtime_helper_023052d0(TypeInfo_FloatSetting);
  Settings_FloatSetting___ctor_40f3a30(pSVar2,15.0,0.0,100.0,(MethodInfo *)0x0);
  (__this->fields).TitanSpawnPunk = pSVar2;
  il2cpp_runtime_helper_022b4080(&(__this->fields).TitanSpawnPunk);
  pSVar1 = (Settings_BoolSetting_o *)il2cpp_runtime_helper_023052d0(TypeInfo_BoolSetting);
  Settings_BoolSetting___ctor_40f3960(pSVar1,0,(MethodInfo *)0x0);
  (__this->fields).TitanSizeEnabled = pSVar1;
  il2cpp_runtime_helper_022b4080(&(__this->fields).TitanSizeEnabled);
  pSVar2 = (Settings_FloatSetting_o *)il2cpp_runtime_helper_023052d0(TypeInfo_FloatSetting);
  Settings_FloatSetting___ctor_40f3a30(pSVar2,1.0,0.1,100.0,(MethodInfo *)0x0);
  (__this->fields).TitanSizeMin = pSVar2;
  il2cpp_runtime_helper_022b4080(&(__this->fields).TitanSizeMin);
  pSVar2 = (Settings_FloatSetting_o *)il2cpp_runtime_helper_023052d0(TypeInfo_FloatSetting);
  Settings_FloatSetting___ctor_40f3a30(pSVar2,3.0,0.1,100.0,(MethodInfo *)0x0);
  (__this->fields).TitanSizeMax = pSVar2;
  il2cpp_runtime_helper_022b4080(&(__this->fields).TitanSizeMax);
  pSVar3 = (Settings_IntSetting_o *)il2cpp_runtime_helper_023052d0(TypeInfo_IntSetting);
  Settings_IntSetting___ctor_40f39a0(pSVar3,0,0,0x7fffffff,(MethodInfo *)0x0);
  (__this->fields).TitanHealthMode = pSVar3;
  il2cpp_runtime_helper_022b4080(&(__this->fields).TitanHealthMode,pSVar3);
  pSVar3 = (Settings_IntSetting_o *)il2cpp_runtime_helper_023052d0(TypeInfo_IntSetting);
  Settings_IntSetting___ctor_40f39a0(pSVar3,100,0,0x7fffffff,(MethodInfo *)0x0);
  (__this->fields).TitanHealthMin = pSVar3;
  il2cpp_runtime_helper_022b4080(&(__this->fields).TitanHealthMin,pSVar3);
  pSVar3 = (Settings_IntSetting_o *)il2cpp_runtime_helper_023052d0(TypeInfo_IntSetting);
  Settings_IntSetting___ctor_40f39a0(pSVar3,200,0,0x7fffffff,(MethodInfo *)0x0);
  (__this->fields).TitanHealthMax = pSVar3;
  il2cpp_runtime_helper_022b4080(&(__this->fields).TitanHealthMax);
  pSVar1 = (Settings_BoolSetting_o *)il2cpp_runtime_helper_023052d0(TypeInfo_BoolSetting);
  Settings_BoolSetting___ctor_40f3960(pSVar1,0,(MethodInfo *)0x0);
  (__this->fields).TitanArmorEnabled = pSVar1;
  il2cpp_runtime_helper_022b4080(&(__this->fields).TitanArmorEnabled);
  pSVar1 = (Settings_BoolSetting_o *)il2cpp_runtime_helper_023052d0(TypeInfo_BoolSetting);
  Settings_BoolSetting___ctor_40f3960(pSVar1,0,(MethodInfo *)0x0);
  (__this->fields).TitanArmorCrawlerEnabled = pSVar1;
  il2cpp_runtime_helper_022b4080(&(__this->fields).TitanArmorCrawlerEnabled,pSVar1);
  pSVar3 = (Settings_IntSetting_o *)il2cpp_runtime_helper_023052d0(TypeInfo_IntSetting);
  Settings_IntSetting___ctor_40f39a0(pSVar3,1000,0,0x7fffffff,(MethodInfo *)0x0);
  (__this->fields).TitanArmor = pSVar3;
  il2cpp_runtime_helper_022b4080(&(__this->fields).TitanArmor);
  pSVar1 = (Settings_BoolSetting_o *)il2cpp_runtime_helper_023052d0(TypeInfo_BoolSetting);
  Settings_BoolSetting___ctor_40f3960(pSVar1,0,(MethodInfo *)0x0);
  (__this->fields).TitanStandardModels = pSVar1;
  il2cpp_runtime_helper_022b4080(&(__this->fields).TitanStandardModels,pSVar1);
  pSVar1 = (Settings_BoolSetting_o *)il2cpp_runtime_helper_023052d0(TypeInfo_BoolSetting);
  Settings_BoolSetting___ctor_40f3960(pSVar1,1,(MethodInfo *)0x0);
  (__this->fields).TitanSmartMovement = pSVar1;
  il2cpp_runtime_helper_022b4080(&(__this->fields).TitanSmartMovement);
  Settings_BaseSettingsContainer___ctor((Settings_PresetSettingsContainer_o *)__this,(MethodInfo *)0x0);
  return;
}

File: csharp_body_deref/Scripts/Settings/InGameTitanSettings.cs
// ==================== AoTTG2 cross-reference ====================
// Type: Settings.InGameTitanSettings
// Update status: unchanged since the previous dump
// Real method bodies (native C decompile): source/cpp/Settings/InGameTitanSettings.c
// Prior real C# source (older reference): Assets/Scripts/Settings/InGame/InGameTitanSettings.cs
// Bodies below are empty IL2CPP metadata stubs; the logic lives in the native C.
// ===============================================================
using Il2CppDummyDll;

namespace Settings;

[Token(Token = "0x2000094")]
internal class InGameTitanSettings : BaseSettingsContainer
{
	[Token(Token = "0x4000350")]
	[FieldOffset(Offset = "0x20")]
	public BoolSetting TitanSpawnEnabled;

	[Token(Token = "0x4000351")]
	[FieldOffset(Offset = "0x28")]
	public FloatSetting TitanSpawnNormal;

	[Token(Token = "0x4000352")]
	[FieldOffset(Offset = "0x30")]
	public FloatSetting TitanSpawnAbnormal;

	[Token(Token = "0x4000353")]
	[FieldOffset(Offset = "0x38")]
	public FloatSetting TitanSpawnJumper;

	[Token(Token = "0x4000354")]
	[FieldOffset(Offset = "0x40")]
	public FloatSetting TitanSpawnCrawler;

	[Token(Token = "0x4000355")]
	[FieldOffset(Offset = "0x48")]
	public FloatSetting TitanSpawnThrower;

	[Token(Token = "0x4000356")]
	[FieldOffset(Offset = "0x50")]
	public FloatSetting TitanSpawnPunk;

	[Token(Token = "0x4000357")]
	[FieldOffset(Offset = "0x58")]
	public BoolSetting TitanSizeEnabled;

	[Token(Token = "0x4000358")]
	[FieldOffset(Offset = "0x60")]
	public FloatSetting TitanSizeMin;

	[Token(Token = "0x4000359")]
	[FieldOffset(Offset = "0x68")]
	public FloatSetting TitanSizeMax;

	[Token(Token = "0x400035A")]
	[FieldOffset(Offset = "0x70")]
	public IntSetting TitanHealthMode;

	[Token(Token = "0x400035B")]
	[FieldOffset(Offset = "0x78")]
	public IntSetting TitanHealthMin;

	[Token(Token = "0x400035C")]
	[FieldOffset(Offset = "0x80")]
	public IntSetting TitanHealthMax;

	[Token(Token = "0x400035D")]
	[FieldOffset(Offset = "0x88")]
	public BoolSetting TitanArmorEnabled;

	[Token(Token = "0x400035E")]
	[FieldOffset(Offset = "0x90")]
	public BoolSetting TitanArmorCrawlerEnabled;

	[Token(Token = "0x400035F")]
	[FieldOffset(Offset = "0x98")]
	public IntSetting TitanArmor;

	[Token(Token = "0x4000360")]
	[FieldOffset(Offset = "0xA0")]
	public BoolSetting TitanStandardModels;

	[Token(Token = "0x4000361")]
	[FieldOffset(Offset = "0xA8")]
	public BoolSetting TitanSmartMovement;

	[Token(Token = "0x6000333")]
	[Address(RVA = "0x3F8BFF0", Offset = "0x3F8BFF0", VA = "0x3F8BFF0")]
	public InGameTitanSettings()
	{
	}
}
